lib/rubocop/cop/sorbet/mixin/target_sorbet_version.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Sorbet
+      module TargetSorbetVersion
+        def self.included(target)
+          target.extend(ClassMethods)
+        end
+
+        module ClassMethods
+          # The version of the Sorbet static type checker required by this cop
+          def minimum_target_sorbet_static_version(version)
+            @minimum_target_sorbet_static_version = Gem::Version.new(version)
+          end
+
+          def support_target_sorbet_static_version?(version)
+            @minimum_target_sorbet_static_version <= Gem::Version.new(version)
+          end
+        end
+
+        def enabled_for_sorbet_static_version?
+          return false unless target_sorbet_static_version_from_bundler_lock_file
+
+          self.class.support_target_sorbet_static_version?(target_sorbet_static_version_from_bundler_lock_file)
+        end
+
+        def target_sorbet_static_version_from_bundler_lock_file
+          @target_sorbet_static_version_from_bundler_lock_file ||= read_sorbet_static_version_from_bundler_lock_file
+        end
+
+        # Adapted from https://github.com/rubocop/rubocop/blob/1181d4ebad5f71c586f9514d9c341cdfffc1957d/lib/rubocop/config.rb#L293-L308
+        def read_sorbet_static_version_from_bundler_lock_file
```

Comment:
This is all pretty yucky. It'd be nice to have a native API to support this

https://github.com/rubocop/rubocop/issues/11920